### PR TITLE
[v2] Add tvOS support

### PR DESCRIPTION
**Pull request by neilkimmett**
Some UIKit classes aren't available on tvOS (they're marked with `__TVOS_PROHIBITED` in the framework headers) so in order to build for tvOS we need to conditionally compile out any references to them. Also adds a tvOS library target in the Xcode project.

### CODE REVIEW

**Comment by neilkimmett** [comment]
Error from CI is 
```
/usr/local/Homebrew/Library/Homebrew/brew.rb:12:in `<main>': Homebrew must be run under Ruby 2.3! (RuntimeError)
Error: No available formula with the name "applesimutils" 
==> Searching for a previously deleted formula...
Warning: homebrew/core is shallow clone. To get complete history run:
  git -C "$(brew --repo homebrew/core)" fetch --unshallow
Error: No previously deleted formula found.
==> Searching for similarly named formulae...
==> Searching local taps...
Error: No similarly named formulae found.
==> Searching taps...
==> Searching taps on GitHub...
Error: No formulae found in taps.
The command "source ./scripts/env/installAppleSimUtils.sh" failed and exited with 1 during .
```
which doesn't seem like its related to my change.

File: lib/ios/RCTHelpers.m [hunk 1]
```diff
@@ -127,7 +127,13 @@ + (NSMutableDictionary *)textAttributesFromDictionary:(NSDictionary *)dictionary
 
 + (NSMutableDictionary *)textAttributesFromDictionary:(NSDictionary *)dictionary withPrefix:(NSString *)prefix
 {
-	return [self textAttributesFromDictionary:dictionary withPrefix:prefix baseFont:[UIFont systemFontOfSize:[UIFont systemFontSize]]];
+#if !(TARGET_OS_TV)
+	CGFloat systemFontSize = [UIFont systemFontSize];
+#else
+	CGFloat systemFontSize = 20;
```
Comment (neilkimmett) [inline lib/ios/RCTHelpers.m]: Worth pointing out I've plucked this default value out of thin air; tvOS doesn't have a `systemFontSize`.

**Comment by gregorymfoster** [comment]
Looks like this is failing `hides Tab Bar when pressing on Hide Top Bar and shows it when pressing on Show Top Bar`, perhaps because the tvos does not support this functionality. Maybe we could modify some of the existing tests to skip the tvos target? 

@gran33 What can we do to get this merged into the main repository?

**Comment by psegalen** [comment]
If someone could merge this it would be great! 🙏

**Comment by yershalom** [comment]
Can one of the admins verify this patch?

**Comment by itsmepetrov** [comment]
Hi @neilkimmett, do you have any plans to continue working on tvOS support? I would like to help you with it.

**Comment by neilkimmett** [comment]
@itsmepetrov hey alas I don't work on that app anymore, and we didn't even end up using this library, we used https://facebook.github.io/react-native/docs/tabbarios. I can help you with any questions you have tho. I gave a talk about that work last year which you might find useful http://kimmett.me/talks/react-native-on-tvos.html

**Pull request closed without merging**